### PR TITLE
Firmware 11

### PR DESCRIPTION
**Pull request by mbonani**
first pull request before intensive testing of binaries

### CODE REVIEW

File: ground_ir.c [hunk 1]
```diff
@@ -28,6 +28,53 @@
...
+
+static int _calib(int value, int i) {
+	int ret;
+	
+	if(value + CALIB_HISTERES > prox_ground_max[i]) {
+		if(++prox_calib_max_counter[i] > 3) {
+			if(value > prox_ground_max[i]) {
+				prox_ground_max[i] = value;
+			} else
+				prox_calib_max_counter[i] = 0;
```
Comment (stephanemagnenat) [inline ground_ir.c]: I would also put a pair of {} around this expression to make it more readable.

File: ground_ir.c [hunk 2]
```diff
@@ -28,6 +28,53 @@
...
+
+static unsigned char prox_calib_max_counter[2];
+static int prox_ground_max[2]; // the calibratoin is not store in settings
+
+static int _calib(int value, int i) {
+	int ret;
+	
+	if(value + CALIB_HISTERES > prox_ground_max[i]) {
+		if(++prox_calib_max_counter[i] > 3) {
+			if(value > prox_ground_max[i]) {
```
Comment (stephanemagnenat) [inline ground_ir.c]: If this is true, `prox_calib_max_counter[i]` is not set to 0, is it what you want?

File: sd.c [hunk 3]
```diff
@@ -190,6 +190,27 @@ void sd_stop_record(void) {
...
+	// Make sure the file is closed
+	f_close(&read_file);
+	if(f_open(&read_file, file, FA_READ) == FR_OK){
+		duration = wav_header_read_duration(&read_file);
+		duration = duration * 5 / 3906; // change in 10th of sec 7812byte/s
+		if ( duration > 32767)
+			duration = 32767;
+	}
+	else 
+		duration=0;
```
Comment (stephanemagnenat) [inline sd.c]: Unconsistent spacing, should be `duration = 0`.

File: wav.c [hunk 4]
```diff
@@ -100,4 +100,23 @@ void wav_finalize_header(FIL * file, unsigned long size) {
...
+}
+
+unsigned long wav_header_read_duration(FIL * file) {
+	unsigned int read;
+	
+	f_read(file, read_buffer, sizeof(read_buffer), &read);
+	if(read != sizeof(read_buffer)) 
+		return 0;
+	
+	if(read_buffer[1] - 36 != read_buffer[10])
```
Comment (stephanemagnenat) [inline wav.c]: Please add comments for these tests, otherwise it is hard to know what the code means.

File: ground_ir.c (hunk 2 above)
Comment (mbonani) [reply ground_ir.c]: yes see lign 45 to 48. We want that 3 continuous value are over the max to update it.

File: wav.c (hunk 4 above)
Comment (mbonani) [reply wav.c]: they are sanity check of the wav header file. I can comment but the rest of the code is not... I want not to instore good habit ;-)

File: ground_ir.c (hunk 1 above)
Comment (stephanemagnenat) [reply ground_ir.c]: Whatever you prefer, just be symmetric.

File: ground_ir.c (hunk 2 above)
Comment (stephanemagnenat) [reply ground_ir.c]: ok, good.

File: wav.c (hunk 4 above)
Comment (stephanemagnenat) [reply wav.c]: ok, then write a small comment above them, something like "sanity checks about the integrity of the wave file".

**Pull request merged**